### PR TITLE
Speed up support

### PR DESCRIPTION
**Pull request by aminlatifi**
Poll smart contract events to see whether the transaction went through or not.
#289

### CODE REVIEW

File: src/blockchain/allowanceHelper.js [hunk 1]
```diff
@@ -4,15 +4,50 @@ import config from '../config';
...
+
+  const clear = () => {
+    if (subscription) {
+      subscription.unsubscribe();
+      subscription = undefined;
+    }
+  };
+
+  return new Promise((resolve, reject) => {
+    ERC20Cotract.approve(config.givethBridgeAddress, amount)
```
Comment (MerlinEgalite) [inline src/blockchain/allowanceHelper.js:17]:
```suggestion
    ERC20Contract.approve(config.givethBridgeAddress, amount)
```

File: src/blockchain/allowanceHelper.js [hunk 2]
```diff
@@ -4,15 +4,50 @@ import config from '../config';
 const INFINITE_ALLOWANCE = new BigNumber(2).pow(256).minus(1).toFixed();
 
 const createAllowance = async (ERC20Cotract, account, amount) => {
```
Comment (MerlinEgalite) [inline src/blockchain/allowanceHelper.js:6]:
```suggestion
const createAllowance = async (ERC20Contract, account, amount) => {
```

File: src/blockchain/allowanceHelper.js [hunk 3]
```diff
@@ -4,15 +4,50 @@ import config from '../config';
...
+      subscription.unsubscribe();
+      subscription = undefined;
+    }
+  };
+
+  return new Promise((resolve, reject) => {
+    ERC20Cotract.approve(config.givethBridgeAddress, amount)
+      .send({ from: account })
+      .on('transactionHash', async transactionHash => {
+        const web3 = ERC20Cotract.getWeb3();
```
Comment (MerlinEgalite) [inline src/blockchain/allowanceHelper.js:20]:
```suggestion
        const web3 = ERC20Contract.getWeb3();
```

File: src/blockchain/allowanceHelper.js [hunk 4]
```diff
@@ -4,15 +4,50 @@ import config from '../config';
...
+        const fromBlock = await web3.eth.getBlockNumber();
+
+        subscription = web3.eth
+          .subscribe('newBlockHeaders')
+          .on('data', async block => {
+            if (!block.number) return;
+            const transactionCount = await web3.eth.getTransactionCount(from);
+
+            if (transactionCount > nonce) {
+              const events = await ERC20Cotract.peApproval({
```
Comment (MerlinEgalite) [inline src/blockchain/allowanceHelper.js:31]:
```suggestion
              const events = await ERC20Contract.peApproval({
```

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]
Just a Typo but everything seems good to me!

**Comment by aminlatifi** [comment]
> Just a Typo but everything seems good to me!

Thanks @MerlinEgalite

**Pull request merged**